### PR TITLE
Add FakeSign.exe execution for non-windows platforms

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

**Comment by tmeschter** [comment]
:+1:

File: build/VSL.Imports.Closed.targets [hunk 1]
```diff
@@ -70,8 +70,8 @@
           Inputs="@(IntermediateAssembly)"
           Outputs="@(IntermediateAssembly);$(PostCompileBinaryModificationSentinelFile)">
     
-    <Exec Command="$(VSLOutDir)\FakeSign.exe &quot;@(IntermediateAssembly)&quot;" />
-    
+    <Exec Condition=" '$(OS)' == 'Windows_NT'" Command="$(VSLOutDir)\FakeSign.exe &quot;@(IntermediateAssembly)&quot;" />
+    <Exec Condition=" '$(OS)' != 'Windows_NT'" Command="mono $(VSLOutDir)\FakeSign.exe &quot;@(IntermediateAssembly)&quot;" />
```
Comment (jmarolf) [inline build/VSL.Imports.Closed.targets]: Is there a way we could check for mono here?  I would like to show a more user friendly message if mono is missing.

**Comment by jaredpar** [comment]
I'm not sure if this is the best approach or not for this problem.  If this approach is taken then we, and every other portable project, will have to change virtually every `Exec` in our MSBuild setup.  That would spread out this logic across a lot of MSBuild files.  

I think perhaps we should look into declaring a new task: NetExec.  Essentially an exec which targets a managed executable that has roughly the logic here: use mono on non-windows platforms, direct execute on windows.

**Comment by tmat** [comment]
+1 for @jaredpar's proposal. We might also want to use CoreClr or other CLI implementations.

**Comment by VSadov** [comment]
+1 for the NetExec. There are other places already where we need to call external managed executables like when syntax and bound tree generators are invoked. Shimming every place individually seems tedious. Indeed, CoreClr or other CLI implementations are likely to use some different "foobar managed.exe". Encapsulating the launcher switching logic in one place should make it easier to use in multiple places.

File: build/VSL.Imports.Closed.targets (hunk 1 above)
Comment (jasonwilliams200OK) [inline build/VSL.Imports.Closed.targets]: @akoeplinger any ideas for this?

Comment (akoeplinger) [inline build/VSL.Imports.Closed.targets]: There is no straightforward way that I'm aware of. Long term I prefer @jaredpar's suggestion of a `NetExec` task.

Comment (jaredpar) [inline build/VSL.Imports.Closed.targets]: I actually tried to implement `NetExec` at one point and it turned out to be significantly more complex than I originally expected.  Eventually I ended up taking the following approach:

https://github.com/dotnet/roslyn/commit/3cef72ec8ed4d8dae6c2210f58ae521927aace6a#diff-9d3884f8609197f7e7500f5d4726077f

It's not nearly as elegant but it works and keeps the number of duplicate elements due to `Conditional` low.

Comment (jasonwilliams200OK) [inline build/VSL.Imports.Closed.targets]: (I am unsure)
Would it make sense to pursue it here: https://github.com/Microsoft/msbuild/blob/82177a50da735cc0443ac10fa490d69368403d71/src/XMakeTasks/Exec.cs?

( curiously, there is [`cmd.exe`](https://github.com/Microsoft/msbuild/blob/xplat/src/XMakeTasks/Exec.cs#L666) in xplat branch as well! :open_mouth: )

Comment (jaredpar) [inline build/VSL.Imports.Closed.targets]: @jasonwilliams200OK i think that would be a valid future option for us.

**Comment by gafter** [comment]
Should we close this PR as one we don't intend to accept? It is quite stale now.

**Pull request closed without merging**